File: tests/src/com.typesafe.akka/akka-protobuf-v3_2.13/2.8.8/src/test/scala/com_typesafe_akka/akka_protobuf_v3_2_13/GeneratedMessageLiteTest.scala
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_typesafe_akka.akka_protobuf_v3_2_13

import akka.protobufv3.internal.GeneratedMessageLiteTestSupport
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

class GeneratedMessageLiteTest {
  @Test
  def getDefaultInstanceInitializesGeneratedMessageClassByName(): Unit = {
    assertTrue(GeneratedMessageLiteTestSupport.lookupDefaultInstanceInitializesGeneratedMessageClass())
  }

  @Test
  def reflectiveGeneratedAccessorHelpersInvokePublicMethod(): Unit = {
    val value: String = GeneratedMessageLiteTestSupport.invokePublicAccessorThroughGeneratedMessageLite()

    assertEquals("invoked through GeneratedMessageLite", value)
  }
}

File: tests/src/com.google.protobuf/protobuf-javalite/4.0.0-rc-1/src/test/java/com_google_protobuf/protobuf_javalite/GeneratedMessageLiteTest.java
/*
 * Copyright and related rights waived via CC0
 *
 * You should have received a copy of the CC0 legalcode along with this
 * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 */
package com_google_protobuf.protobuf_javalite;

import static org.assertj.core.api.Assertions.assertThat;

import com.google.protobuf.GeneratedMessageLite;
import com.google.protobuf.StringValue;
import java.lang.reflect.Method;
import org.junit.jupiter.api.Test;

public class GeneratedMessageLiteTest {
    @Test
    void toStringPrintsLiteMessageFieldsThroughGeneratedAccessors() {
        StringValue message = StringValue.of("generated-message-lite");

        String textFormat = message.toString();

        assertThat(textFormat)
                .startsWith("# com.google.protobuf.StringValue@")
                .contains("value: \"generated-message-lite\"");
    }

    @Test
    void generatedAccessorLookupFindsPublicDefaultInstanceMethod() throws Exception {
        Method helper = GeneratedMessageLite.class.getDeclaredMethod(
                "getMethodOrDie", Class.class, String.class, Class[].class);
        helper.setAccessible(true);

        Method defaultInstanceMethod = (Method) helper.invoke(
                null, StringValue.class, "getDefaultInstance", new Class<?>[0]);

        assertThat(defaultInstanceMethod.getDeclaringClass()).isEqualTo(StringValue.class);
        assertThat(defaultInstanceMethod.getName()).isEqualTo("getDefaultInstance");
        assertThat(defaultInstanceMethod.invoke(null)).isSameAs(StringValue.getDefaultInstance());
    }
}
